rts/lra/coordinator/src/main/java/io/narayana/lra/coordinator/domain/model/LongRunningAction.java

```diff
@@ -397,8 +410,13 @@ protected ReentrantLock tryLockTransaction() {
     public int finishLRA(boolean cancel) {
         ReentrantLock lock = null;
 
-        // remove the LRA from the timeout scheduler (no need to take the lock)
-        TransactionReaper.transactionReaper().remove(this);
+        // check whether the transaction should cancel due to a timeout:
+        if (finishTime != null && !cancel && ChronoUnit.MILLIS.between(LocalDateTime.now(ZoneOffset.UTC), finishTime) <= 0) {
```

Comment:
This sounds good. As it makes the timeout processing stable it's a great fix.

Reply:
Thanks, simple fixes are so much sweater.